frameit/lib/frameit/template_finder.rb

```diff
@@ -1,38 +1,82 @@
 module Frameit
   # Responsible for finding the correct device
   class TemplateFinder
+
+    class FilenameTransform
+      def initialize(device_name)
+        @device_name = device_name
+      end
+    end
+
+    # Example: iPhone_5s_Vert_SpaceGray_sRGB
+    class Type1Transform < FilenameTransform
+      def transform(color, orientation)
+        # Note the * on the end to handle the _sRGB part
+        "#{@device_name}_#{orientation}_#{color}*"
+      end
+    end
+
+    # Example: iPad-Pro-Space-Gray-vertical.png
+    class Type2Transform < FilenameTransform
+      def transform(color, orientation)
+        fixed_color = color == 'SpaceGray' ? "Space-Gray" : "Silver"
+        fixed_orientation = orientation == 'Horz' ? 'horizontal' : 'vertical'
+        "#{@device_name}-#{fixed_color}-#{fixed_orientation}"
+      end
+    end
+
+    # Example: iPhone-SE-Space-Gray
+    # Note that 'vertical' is implied
+    class Type3Transform < FilenameTransform
+      def transform(color, orientation)
+        fixed_color = color == 'SpaceGray' ? "Space-Gray" : "Silver"
+        filename = "#{@device_name}-#{fixed_color}"
+        if orientation == 'Horz'
+          "#{filename}-horizontal"
+        else
+          filename
+        end
+      end
+    end
+
+    DEVICE_TO_TRANSFORM_TYPE = {
+      'iPhone-SE' => Type3Transform.new('iPhone-SE'),
+      'iPad-Pro' => Type2Transform.new('iPad-Pro'),
+      'iPad-mini' => Type2Transform.new('iPad-mini'),
+      'iPhone-6s' => Type2Transform.new('iPhone-6s'),
+      'iPhone-6s-Plus' => Type2Transform.new('iPhone-6s-Plus')
+    }
+
     # This will detect the screen size and choose the correct template
     def self.get_template(screenshot)
       return nil if screenshot.mac?
-      parts = [
-        screenshot.device_name,
-        screenshot.orientation_name,
-        screenshot.color
-      ]
-      joiner = "_"
-
-      if screenshot.device_name.include?('iPad') || screenshot.device_name.include?('6s')
-        parts = [
-          screenshot.device_name,
-          (screenshot.color == 'SpaceGray' ? "Space-Gray" : "Silver"),
-          (screenshot.orientation_name == "Horz" ? "horizontal" : "vertical")
-        ]
-        joiner = "-"
-      end
+
+      transformer = DEVICE_TO_TRANSFORM_TYPE[screenshot.device_name]
+
+      transformer ||= Type1Transform.new(screenshot.device_name)
```

Comment:
In what cases would we fail to find a proper transform in our lookup table? Why is falling back to a Type1 correct, rather than raising an error? Can that rationale be documented?